pub/lib/dependabot/pub/helpers.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+require "json"
+require "open3"
+require "digest"
+
+require "dependabot/errors"
+require "dependabot/shared_helpers"
+require "dependabot/pub/requirement"
+
+module Dependabot
+  module Pub
+    module Helpers
+      private
+
+      def dependency_services_list
+        JSON.parse(run_dependency_services("list"))["dependencies"]
+      end
+
+      def dependency_services_report
+        sha256 = Digest::SHA256.new
+        dependency_files.each do |f|
+          sha256 << f.path + "\n" + f.content + "\n"
+        end
+        hash = sha256.hexdigest
+
+        cache_file = "/tmp/report-#{hash}-pid-#{Process.pid}.json"
```

Comment:
@dependabot/reviewers: Is this the right spot to maintain a cache that persists across dependency updates within an update job? This looks like it would give a performance boost to version updates by not having to re-resolve this for each dependency whereas `SharedHelpers.in_a_temporary_directory` would get reset between each dependency.

Reply:
Yeah I think so, since the folder is owned by `root` I was worried the `dependabot` user might not have write access, but looks like it does:

```
w[dependabot-core-dev] ~/dependabot-core $ whoami
dependabot
[dependabot-core-dev] ~/dependabot-core $ irb
irb(main):001:0> File.write("/tmp/permissions-test", "testing 123, is this thing on?")
=> 30
irb(main):002:0> exit
[dependabot-core-dev] ~/dependabot-core $ cat /tmp/permissions-test
testing 123, is this thing on?
```

👍